src/types/responses/Paras.ts

```diff
@@ -0,0 +1,170 @@
+import { AbstractInt } from '@polkadot/types/codec/AbstractInt';
+import {
+	AccountId,
+	BalanceOf,
+	FundInfo,
+	ParaId,
+	ParaLifecycle,
+} from '@polkadot/types/interfaces';
+import BN from 'bn.js';
+
+import { IOption } from '../util';
+import { IAt } from './';
+
+export type AuctionPhase = 'starting' | 'ending';
+
+export type ParaType = 'parachain' | 'parathread';
+
+export interface IFund {
+	/**
+	 * Id of the parachain or parathread that has a crowdloan.
+	 */
+	paraId: ParaId;
+	/**
+	 * `FundInfo` for the para's crowdloan. Use query param `fundInfo=true` to include the `FundInfo`.
+	 * Inclusion may make the query take longer.
+	 */
+	fundInfo?: Option<FundInfo>;
+}
+
+export interface ICrowdloansInfo {
+	at: IAt;
+	/**
+	 * `FundInfo` describing the crowdloan, or null if none could be found.
+	 */
+	fundInfo: IOption<FundInfo>;
+	/**
+	 * Lease period indexes that the crowdloan applys to.
+	 */
+	leasePeriods?: number[];
+}
+
+export interface ICrowdloans {
+	at: IAt;
+	/**
+	 * List of all crowdloans.
+	 */
+	funds: IFund[];
+}
+
+export interface LeaseFormatted {
+	/**
+	 * Lease period.
+	 */
+	leasePeriodIndex: number;
+	/**
+	 * Amount held on deposit for the lease period.
+	 */
+	deposit: BalanceOf;
+	/**
+	 * Account responsible for the lease.
+	 */
+	account: AccountId;
+}
+
+export interface ILeaseInfo {
+	at: IAt;
+	/**
+	 * Lifecycle of the para (i.e Onboarding, Parathread, Offboarding etc).
+	 */
+	paraLifecycle: Option<ParaLifecycle>;
+	/**
+	 * If the para is in the onboarding phase, this will say if it is onboarding as
+	 * a `parachain` or a `parathread`; otherwise it this field is not included in the response.
+	 */
+	onboardingAs?: ParaType;
+	/**
+	 * List of current and upcoming leases this para has.
+	 */
+	leases: IOption<LeaseFormatted[]>;
+}
+
+/**
+ * `auctions::WinningData` expressed as an object.
+ */
+export interface IWinningData {
+	accountId: AccountId;
+	paraId: ParaId;
+	amount: BalanceOf;
+}
+
+/**
+ * Bid and its corresponding set of leases.
+ */
+export interface IWinningDataWithLeaseSet {
+	bid: IOption<IWinningData>;
+	leaseSet: number[];
+}
+
+export interface IAuctionsCurrent {
+	at: IAt;
+	/**
+	 * Fist block of the auction ending phase.
+	 */
+	beginEnd: IOption<AbstractInt>;
+	/**
+	 * Last block of the auction ending phase.
+	 */
+	finishEnd: IOption<BN>;
+	/**
+	 * Phase of auction. One of `opening` or `ending`. The `ending` phase is where
+	 * an eventual winner is chosen retroactively by randomly choosing a block number
+	 * in the `Ending` phase and using the `winning` bids.
+	 */
+	phase: IOption<AuctionPhase>;
+	/**
+	 * Auction number. If there is no current auction this will be the number of
+	 * the previous auction.
+	 */
+	auctionIndex: AbstractInt;
+	/**
+	 * Lease period indexes that may be bid on in this auction.
+	 */
+	leasePeriods: IOption<number[]>;
+	/**
+	 * Winning bids at this current block height. Is only not `null` during the
+	 * `Ending` phase.
```

Comment:
```suggestion
	 * Winning bids at this current block height. Always `null` outside the
	 * `Ending` phase.
```

Reply:
I actually need to correct this. It is `null` when there are no bids. But once there is a single bid (at any phase in the auction) this will be an array.